internal/services/cognitive/cognitive_account_rai_policy_resource.go

```diff
@@ -0,0 +1,396 @@
+	return map[string]*pluginsdk.Schema{
+		"type": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+	}
+}
+
+func (r CognitiveAccountRaiPolicyResource) ModelObject() interface{} {
+	return &AccountRaiPolicyResourceModel{}
+}
+
+func (r CognitiveAccountRaiPolicyResource) ResourceType() string {
+	return "azurerm_cognitive_account_rai_policy"
+}
+
+func (r CognitiveAccountRaiPolicyResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.Cognitive.RaiPoliciesClient
+			subscriptionId := metadata.Client.Account.SubscriptionId
+
+			var model AccountRaiPolicyResourceModel
+			if err := metadata.Decode(&model); err != nil {
+				return err
+			}
+
+			cognitiveAccountId, err := raipolicies.ParseAccountID(model.AccountId)
+			if err != nil {
+				return err
+			}
+
+			id := raipolicies.NewRaiPolicyID(subscriptionId, cognitiveAccountId.ResourceGroupName, cognitiveAccountId.AccountName, model.Name)
+			existing, err := client.Get(ctx, id)
+			if err != nil && !response.WasNotFound(existing.HttpResponse) {
+				return fmt.Errorf("checking for presence of existing %s: %+v", id, err)
+			}
+			if !response.WasNotFound(existing.HttpResponse) {
+				return metadata.ResourceRequiresImport(r.ResourceType(), id)
+			}
+
+			locks.ByID(cognitiveAccountId.ID())
+			defer locks.UnlockByID(cognitiveAccountId.ID())
+
+			raiPolicy := raipolicies.RaiPolicy{
+				Name: pointer.To(model.Name),
+				Properties: &raipolicies.RaiPolicyProperties{
+					BasePolicyName:   pointer.To(model.BasePolicyName),
+					ContentFilters:   expandRaiPolicyContentFilters(model.ContentFilter),
+					CustomBlocklists: expandCustomBlockLists(model.CustomBlockList),
+					Mode:             pointer.To(raipolicies.RaiPolicyMode(model.Mode)),
+				},
+				Tags: pointer.To(model.Tags),
+			}
+
+			if _, err := client.CreateOrUpdate(ctx, id, raiPolicy); err != nil {
+				return fmt.Errorf("creating %s: %+v", id, err)
+			}
+
+			metadata.SetID(id)
+
+			return nil
+		},
+	}
+}
+
+func (r CognitiveAccountRaiPolicyResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.Cognitive.RaiPoliciesClient
+
+			id, err := raipolicies.ParseRaiPolicyID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			cognitiveAccountId := raipolicies.NewAccountID(id.SubscriptionId, id.ResourceGroupName, id.AccountName)
+
+			resp, err := client.Get(ctx, *id)
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					return metadata.MarkAsGone(id)
+				}
+				return fmt.Errorf("retrieving %s: %+v", id, err)
+			}
+
+			state := AccountRaiPolicyResourceModel{}
+
+			if model := resp.Model; model != nil {
+				state.Name = pointer.From(model.Name)
+				state.AccountId = cognitiveAccountId.ID()
```

Comment:
We should get the name from the ID not from the model, also these should be set outside of nil checking `resp.Model` since they don't depend on it
```suggestion
			state := AccountRaiPolicyResourceModel{
				Name: id.Name,
				AccountId: cognitiveAccountId.ID(),
			}

			if model := resp.Model; model != nil {
```